include/mbedtls/gcm.h

```diff
@@ -287,13 +336,26 @@ int mbedtls_gcm_update( mbedtls_gcm_context *ctx,
  *                  buffer of at least \p tag_len Bytes.
  * \param tag_len   The length of the tag to generate. This must be at least
  *                  four.
+ * \param output    The buffer for the final output.
+ *                  If \p output_size is nonzero, this must be a writable
+ *                  buffer of at least \p output_size bytes.
+ * \param output_size  The size of the \p output buffer in bytes.
+ *                  This must be large enough for the output that
+ *                  mbedtls_gcm_update() has not produced. In particular:
+ *                  - If mbedtls_gcm_update() produces immediate output,
+ *                    or if the total input size is a multiple of \c 16,
+ *                    then mbedtls_gcm_finish() never produces any output,
+ *                    so \p output_size can be \c 0.
+ *                  - \p output_size never needs to be more than \c 15.
```

Comment:
Missing: an `output_length` parameter. It's possible for the caller to know how many bytes of output to expect, since the total length of the plaintext must be equal to the total length of the ciphertext. But that requires the caller to keep track of the cumulated length of input and output from update calls, which would be impractical, for example, in the library's own psa_crypto_aead and cipher modules.